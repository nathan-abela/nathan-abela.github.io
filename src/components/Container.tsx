import styled from 'styled-components';
import {
    alignContent,
    alignItems,
    borders,
    color,
    compose,
    display,
    flexDirection,
    grid,
    justifyContent,
    justifyItems,
    margin,
    padding,
    textAlign,
    layout,
    AlignContentProps,
    AlignItemsProps,
    BordersProps,
    ColorProps,
    DisplayProps,
    FlexDirectionProps,
    GridProps,
    JustifyContentProps,
    JustifyItemsProps,
    LayoutProps,
    MarginProps,
    PaddingProps,
    TextAlignProps
} from 'styled-system';

export type ContainerProps = AlignContentProps &
    AlignItemsProps &
    BordersProps &
    ColorProps &
    DisplayProps &
    FlexDirectionProps &
    GridProps &
    JustifyContentProps &
    JustifyItemsProps &
    LayoutProps &
    MarginProps &
    PaddingProps &
    TextAlignProps;

const Container = styled.div<ContainerProps>`
    display: flex;
    flex-direction: column;
    flex-wrap: wrap;

    ${compose(
    alignContent,
    alignItems,
    borders,
    color,
    display,
    flexDirection,
    grid,
    justifyContent,
    justifyItems,
    layout,
    margin,
    padding,
    textAlign
)};
`;

export default Container;
