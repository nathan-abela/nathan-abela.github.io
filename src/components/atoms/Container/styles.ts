import styled from 'styled-components';

const StyledContainer = styled.div`
	width: 94%;
	max-width: 880px;
	margin: 0 auto;
`;

export { StyledContainer };
