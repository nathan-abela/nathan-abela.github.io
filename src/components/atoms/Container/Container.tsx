import { FC } from 'react';
import { StyledContainer } from './styles';

interface ContainerProps {
	children: React.ReactNode;
}

const Container: FC<ContainerProps> = ({ children }) => {
	return <StyledContainer>{children}</StyledContainer>;
};

export { Container };
