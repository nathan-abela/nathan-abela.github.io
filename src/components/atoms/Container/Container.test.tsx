import { render } from '@testing-library/react';

import { Container } from './Container';

describe('Container', () => {
	it('renders container correctly', () => {
		const { container } = render(
			// eslint-disable-next-line prettier/prettier
			<Container>Page Content</Container>
		);
		expect(container).toMatchSnapshot();
	});
});
