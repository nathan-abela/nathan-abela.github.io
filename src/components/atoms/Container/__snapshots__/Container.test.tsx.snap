// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container renders container correctly 1`] = `
<div>
  <div
    class="styles__StyledContainer-sc-a6yh8x-0 earCTD"
  >
    Page Content
  </div>
</div>
`;

exports[`Container renders correctly 1`] = `
<div>
  <div
    class="styles__StyledContainer-sc-a6yh8x-0 earCTD"
  >
    Page Content
  </div>
</div>
`;
